src/Migration/Sources/Csv.php

```diff
@@ -0,0 +1,338 @@
+use Utopia\Migration\Resource;
+use Utopia\Migration\Resources\Database\Attribute;
+use Utopia\Migration\Resources\Database\Collection;
+use Utopia\Migration\Resources\Database\Database;
+use Utopia\Migration\Resources\Database\Document;
+use Utopia\Migration\Resources\Storage\File;
+use Utopia\Migration\Source;
+use Utopia\Migration\Sources\Appwrite\Reader;
+use Utopia\Migration\Sources\Appwrite\Reader\Database as DatabaseReader;
+use Utopia\Migration\Transfer;
+use Utopia\Storage\Device;
+
+class Csv extends Source
+{
+    private string $filePath;
+
+    /**
+     * format: `{databaseId:collectionId}`
+     */
+    private string $resourceId;
+
+    private Device $deviceForCsvImports;
+
+    private Reader $database;
+
+    private ?UtopiaDatabase $dbForProject;
+
+    public function __construct(
+        string $resourceId,
+        string $filePath,
+        Device $deviceForCsvImports,
+        ?UtopiaDatabase $dbForProject
+    ) {
+        $this->filePath = $filePath;
+        $this->resourceId = $resourceId;
+        $this->deviceForCsvImports = $deviceForCsvImports;
+
+        $this->dbForProject = $dbForProject;
+        $this->database = new DatabaseReader($dbForProject);
+    }
+
+    public static function getName(): string
+    {
+        return 'Csv';
+    }
+
+    public static function getSupportedResources(): array
+    {
+        return [
+            Resource::TYPE_DOCUMENT,
+        ];
+    }
+
+    public function report(array $resources = []): array
+    {
+        return [];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function exportGroupAuth(int $batchSize, array $resources): void
+    {
+        throw new \Exception('Not Implemented');
+    }
+
+    protected function exportGroupDatabases(int $batchSize, array $resources): void
+    {
+        try {
+            if (\in_array(Resource::TYPE_DOCUMENT, $resources)) {
+                $this->exportDocuments($batchSize);
+            }
+        } catch (\Throwable $e) {
+            $this->addError(
+                new Exception(
+                    Resource::TYPE_DOCUMENT,
+                    Transfer::GROUP_DATABASES,
+                    message: $e->getMessage(),
+                    code: $e->getCode(),
+                    previous: $e
+                )
+            );
+        } finally {
+            // delete the temporary file!
+            $this->deviceForCsvImports->delete($this->filePath);
+        }
+    }
+
+    /**
+     * @throws \Exception
+     */
+    private function exportDocuments(int $batchSize): void
+    {
+        if (! $this->deviceForCsvImports->exists($this->filePath)) {
+            return;
+        }
+
+        $stream = fopen($this->filePath, 'r');
+        if (! $stream) {
+            return;
+        }
+
+        $headers = fgetcsv($stream);
+        if (! is_array($headers) || count($headers) === 0) {
+            fclose($stream);
+            return;
+        }
+
+        $allAttributes = [];
```

Comment:
Let's name this just `attributes` (you're gonna learn I like short names :sweat_smile:)